internal/kinds/capp/utils/route.go

```diff
@@ -0,0 +1,66 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	zoneCM          = "dns-zone"
+	zoneKey         = "zone"
+	placeholderZone = "capp.com."
+	dot             = "."
+)
+
+// GetZoneFromConfig returns the zone to be used for the record from a ConfigMap.
+func GetZoneFromConfig(ctx context.Context, k8sClient client.Client) (string, error) {
+	var zone string
+	var ok bool
+
+	routeCM := corev1.ConfigMap{}
+	if err := k8sClient.Get(ctx, types.NamespacedName{Namespace: CappNS, Name: zoneCM}, &routeCM); err != nil {
+		return "", fmt.Errorf("could not fetch configMap %q from namespace %q: %w", zoneCM, CappNS, err)
+	}
+
+	if len(routeCM.Data) > 0 {
+		zone, ok = routeCM.Data[zoneKey]
+		if !ok {
+			return zone, fmt.Errorf("%q key is not set in ConfigMap %q", zoneKey, zoneCM)
+		} else if zone == "" {
+			return zone, fmt.Errorf("%q is empty in ConfigMap %q", zoneKey, zoneCM)
+		} else if !strings.HasSuffix(zone, dot) {
+			return zone, fmt.Errorf("%q value must end with a %q in ConfigMap %q", zoneKey, dot, zoneCM)
+		}
+	} else {
+		zone = placeholderZone
+	}
+
+	return zone, nil
+}
+
+// ComputeResourceName adjusts the hostname based on the provided suffix and a dot(".") trailing character.
+// If the hostname does not already end with the suffix (minus the trailing dot), it appends the suffix to the hostname.
+func ComputeResourceName(hostname, suffix string) string {
+	suffixWithoutTrailingChar := suffix[:len(suffix)-len(dot)]
+	if strings.HasSuffix(hostname, suffixWithoutTrailingChar) {
```

Comment:
change the if to not if  and change host  name and in the end of the function return the hostname